lib/aur-chroot

```diff
@@ -46,7 +46,7 @@ while true; do
         -C|--pacman-conf)
             shift; pacman_conf=$1 ;;
         -D|--directory)
-            shift; directory=$1 ;;
+            shift; directory="$1/aurbuild/$machine" ;;
```

Comment:
Initially I was wondering if we really needed to force the `.../aurbuild/$machine` structure when the user is passing the path explicitly, but `mkarchroot` is really picky about the workdir arg you pass. It's an ok trade off vs massaging the path inside prepare.

Reply:
I don't like this approach of hardcoding paths - where does `mkarchroot` have any requirements that a path should look like this?

Reply:
Both `aurbuild` and prev `aur-chroot` have had an extra path component appended to one provided with `-D`. I thought this behavior is nice to keep for easy cleanup and backward compatibility with previous implementations.

But if you want `mkarchroot` to work directly in the path provided with `-D` flag, we need to adjust the `chroot prepare` section, currently it fires only if `$directory` doesn't exist.
We will end up with something like this:
```diff
diff --git a/lib/aur-build b/lib/aur-build
index eeb0f42..2d45ac1 100755
--- a/lib/aur-build
+++ b/lib/aur-build
@@ -106,7 +106,7 @@ while true; do
             repo_add_args+=(-s) ;;
         # chroot options
         -D|--directory)
-            shift; makechrootpkg_args+=(-r "$1") ;;
+            shift; chroot_args+=(-D "$1") ;;
         --bind)
             shift; makechrootpkg_args+=(-D "$1") ;;
         --bind-rw)
diff --git a/lib/aur-chroot b/lib/aur-chroot
index 21935f2..17ebdc5 100755
--- a/lib/aur-chroot
+++ b/lib/aur-chroot
@@ -90,7 +90,9 @@ if (( prepare )); then
     # parent path is not created by mkarchroot (#371)
     if [[ ! -d $directory ]]; then
         sudo install -d "$directory" -m 755 -v
+    fi
 
+    if [[ ! -d $directory/root ]]; then
         sudo mkarchroot -C "$pacman_conf" -M "$makepkg_conf" \
              "$directory"/root "${base_packages[@]}"
     fi
```

Reply:
> where does `mkarchroot` have any requirements that a path should look like this?

It requires that all path components must exist except the last. (`readlink -f`). Which doesn't allow us just rely on it by simply give it "$directory/root", we would need the multiple checks mentioned in the comment above.

@bartoszek has mentioned the good points of keeping at least an extra level hardcoded. Other minor ones could include the namescheme from arch-nspawn lock files being quite simple (user.lock) having the extra layer could help avoid collisions, especially when used directly in /tmp which I assume is a common use case for this flag.

Reply:
@rafasc One more thing came to my mind. We might also check if the `$directory` mount point has `nosuid` flag set up. This prohibits `mkchrootpkg` to run but not `mkarchroot`.
```sh
if [[ $(findmnt -n -o OPTIONS --target "$directory") =~ nosuid ]]; then
  error "Unable to build inside \"%s\" as it's mounted with \"nosuid\" option." $(findmnt -n -o TARGET --target "$directory")
  exit 2
fi
```

Reply:
Let the code fail naturally.

Reply:
I don't like all this massaging. :unamused:
The way this is going, i'd rather tell users to use <code>-D /tmp/<em>something</em></code>

Reply:
I guess the proposed change of 
```
     if [[ ! -d $directory ]]; then
         sudo install -d "$directory" -m 755 -v
+    fi
 
+    if [[ ! -d $directory/root ]]; then
         sudo mkarchroot -C "$pacman_conf" -M "$makepkg_conf" \
              "$directory"/root "${base_packages[@]}"
     fi
```
is reasonable. It follows the mentioned simple naming scheme (`$USER`, root) and doesn't hardcode `.arch-chroot` like in 2.3.